CMakeLists.txt

```diff
@@ -1,64 +1,75 @@
-cmake_minimum_required(VERSION 3.9.5)
+cmake_minimum_required(VERSION 3.8.0)
 
-project(libzippp)
+project(libzippp VERSION 0.2.0)
 
-set(PACKAGE "libzippp")
-set(VERSION "0.2")
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    set(is_root_project ON)
+else()
+    set(is_root_project OFF)
+endif()
```

Comment:
@Flamefire Since I'm far from being a CMake expert, could you improve it to add some comments ? :pray:

Reply:
Great! That makes for a good test-drive. I'd rather not add comments because "comments are always wrong or outdated". I made the CMake so it is as readable as possible and should be understandable even by non-experts. Can you check to see if there is anything in particular which you cannot understand?

Reply:
@Flamefire Mostly this actually :sweat_smile: the options `LIBZIPPP_INSTALL` and `LIBZIPPP_BUILD_TESTS` are activated if the project is "root". Does it mean that the compilation is different is I make something like:
```sh
cmake .
make
```
Than if a I do:
```sh
mkdir build && cd build
cmake ..
make
```
Maybe can you also update the README to include those options ? :pray:

Reply:
No the difference is: The folder passed to `cmake` is the `CMAKE_SOURCE_DIR`, in both your examples this is the same folder. When using libzippp via `add_subdirectory` to build it alongside another project, then `CMAKE_SOURCE_DIR` would be the folder of that project (which would then be the root project and libzippp a sub project).

I hope the new comment makes the intention clear